v3-nginx/docker-entrypoint.d/90-nginx-extras-envsubst.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+# vim:sw=2:ts=2:sts=2:et
+
+set -eu
+
+LC_ALL=C
+ME=$( basename "$0" )
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+export DNS_SERVER="${DNS_SERVER:-$(grep -i '^nameserver' /etc/resolv.conf |head -n1|cut -d ' ' -f2)}"
+
+touch /etc/nginx/nginx.conf 2>/dev/null || { echo >&2 "$ME: error: can not modify /etc/nginx/nginx.conf (read-only file system?)"; exit 0; }
+
+nginx_base_envsubst() {
+  local template="/etc/nginx/nginx.conf.template"
+  local suffix=".template"
+  local output_dir="/etc/nginx"
+
+  local template defined_envs relative_path output_path subdir
+  defined_envs=$(printf '${%s} ' $(env | cut -d= -f1))
+  if [ ! -w "$output_dir" ]; then
+    echo >&3 "$ME: ERROR: $output_dir is not writable"
+    return 0
+  fi
+
+  relative_path="${template#$output_dir/}"
+  output_path="$output_dir/${relative_path%$suffix}"
+  subdir=$(dirname "$relative_path")
+  echo >&3 "$ME: Running envsubst on $template to $output_path"
```

Comment:
Where are you writing to? Which file is `3`?

Reply:
This came from upstream, but removed in new iteration.